### DISCUSSION TITLE
Generate Error Messages

### DISCUSSION THREAD

**Discussion by MitchCayCRC** [JoelBender/bacpypes#419] [Q&A]
I have a virtual device that I want to generate error messages for from time to time but don't want to change the properties of the objects.

For example lets say I have a property that is mutable =True, but under certain conditions it should not be so. I want the user to be able to attempt to write to the object (it should be allowed because mutable = True) but after checking for conditions I want to return an error. There is no iocb.ioError so I want to 'generate'/display an error that I created.

Is there a feasible way of doing this?

**Comment by JoelBender** [comment]
I understand that under certain conditions you would like the request to return an error rather than be accepted.  Try [this gist](https://gist.github.com/JoelBender/bd5318907a1d7caabd39eb2d577a7335) and see if that's what you are trying to accomplish.

**Reply by MitchCayCRC** [reply]
Is that the only place where it is possible to display errors to a user? I tried raising ExecutionErrors in other parts of my code but those messages are only displayed on the backend in the kernel as opposed to the user if they were using something such as Yabe

**Comment by JoelBender** [comment]
Ah, I think I misread your original request.  As a library designed to just support device-to-device communications using the BACnet protocol, there is no "user".  If Yabe supports event notifications that can include descriptive text (see **13.8 ConfirmedEventNotification Service**) then the application could be updated to send an event notification when the value goes into "fault".

If you would like to have a user interface for this application (like maybe a web server that displays a Gitter/Slack conversation window? or build in an IRC server and connect using a client like [HexChat](https://hexchat.github.io/)?) that is beyond the scope of this library.

**Comment by MitchCayCRC** [comment]
No worries Joel, I have not been very clear in what I am attempting to achieve. I essentially am creating a device to replicate/mirror another device. 

- Lets call the device I am creating device 1 and the device it is copying device 2. 
- Say device 2 has an object with a writable property but the range of that property is 1-10, the mirrored object on device 1 is unaware of this range limitation
- a write property request is made to device 1 to write a value of 11
- a new request is created and sent along to device 2, which returns an iocb.ioError
- at this point I would like to pass the error along to device 1 but this is where I seem to be stuck. I can raise an ExceptionError but am unsure if this is actually what I am trying to accomplish.
